packages/run-protocol/src/interchainPool.js

```diff
@@ -0,0 +1,138 @@
+// @ts-check
+import { E, Far } from '@endo/far';
+import { AmountMath, AssetKind, makeIssuerKit } from '@agoric/ertp';
+import { offerTo } from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X, quote: q } = assert;
+
+const COSMOS_DECIMALS = 6;
+
+/**
+ * Given sufficient IST, create an issuer for an IBC denom
+ * and an invitation to add a pool for it to the AMM.
+ *
+ * @param {ZCF<InterchainPoolTerms>} zcf
+ * @param {{
+ *   bankManager: ERef<BankManager>,
+ * }} pa
+ *
+ * @typedef {{
+ *   minimumCentral: Amount<'nat'>,
+ *   amm: ERef<XYKAMMPublicFacet>,
+ * }} InterchainPoolTerms
+ *
+ * @typedef {{
+ *   denom: string,
+ *   decimalPlaces: number,
+ * }} AssetDetail
+ */
+export const start = (zcf, { bankManager }) => {
+  const {
+    brands: { Central: centralBrand },
+    minimumCentral,
+    amm,
+  } = zcf.getTerms();
+  AmountMath.coerce(centralBrand, minimumCentral);
+
+  let kwNonce = 0;
+
+  /**
+   * @param {ZCFSeat} seat
+   * @param {AssetDetail} detail
+   */
+  const step1Handler = async (seat, detail) => {
+    assert.typeof(
+      detail,
+      'object',
+      'offer arg must be { denom, decimalPlaces? }',
+    );
+    const { denom, decimalPlaces = COSMOS_DECIMALS } = detail;
+    assert.typeof(denom, 'string');
+    assert.typeof(decimalPlaces, 'number');
+
+    const {
+      give: { Central: centralAmt },
+    } = seat.getProposal();
+    assert(
+      AmountMath.isGTE(centralAmt, minimumCentral),
+      X`at least ${q(minimumCentral)} required; only ${q(centralAmt)} given`,
+    );
+
+    const interAsset = makeIssuerKit(
+      denom,
+      AssetKind.NAT,
+      harden({ decimalPlaces }),
+    );
+    kwNonce += 1;
+
+    const liquidityIssuer = await E(amm).addPool(
```

Comment:
> @Chris-Hibbert the addPool call would move to step 2 when we require funds (https://github.com/Agoric/agoric-sdk/pull/5377). I can't move it yet because the AMM requires the liquidity brand in the want when adding liquidity, even though it doesn't use it.

> Maybe I should fix that in this PR?

Before and after #5377 the APIs for adding a pool are just different. Our choices are to reconcile in whichever PR is second, or merge this into #5377 once approved.

Reply:
I did fix the odd AMM constraint and move `addPool` to step 2. That way, when #5377 lands, we'll have the collateral on hand when we want to `addPool`.